scala/tempo/src/main/scala/com/databrickslabs/tempo/ml/stat/SMA.scala

```diff
@@ -0,0 +1,288 @@
+	 */
+	def apply(measureCol: String,
+	          outputCol: String,
+	          size: Long,
+	          ordering: Ordering,
+	          alignment: WindowAlignment,
+	          includesNow: Boolean): SMA =
+		new SMA().setMeasureCol(measureCol)
+		         .setOutputCol(outputCol)
+		         .setWindowSize(size)
+		         .setWindowOrdering(ordering)
+		         .setWindowAlignment(alignment)
+		         .setIncludesNow(includesNow)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param size
+	 * @param ordering
+	 * @param alignment
+	 * @param includesNow
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          size: Long,
+	          ordering: Ordering,
+	          alignment: WindowAlignment,
+	          includesNow: Boolean): SMA =
+		new SMA().setMeasureCol(measureCol)
+		         .setWindowSize(size)
+		         .setWindowOrdering(ordering)
+		         .setWindowAlignment(alignment)
+		         .setIncludesNow(includesNow)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param outputCol
+	 * @param size
+	 * @param ordering
+	 * @param alignment
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          outputCol: String,
+	          size: Long,
+	          ordering: Ordering,
+	          alignment: WindowAlignment): SMA =
+		apply(measureCol,outputCol,size,ordering,alignment,true)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param size
+	 * @param ordering
+	 * @param alignment
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          size: Long,
+	          ordering: Ordering,
+	          alignment: WindowAlignment): SMA =
+		apply(measureCol,size,ordering,alignment,true)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param outputCol
+	 * @param size
+	 * @param ordering
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          outputCol: String,
+	          size: Long,
+	          ordering: Ordering): SMA =
+		apply(measureCol,outputCol,size,ordering,trailing)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param size
+	 * @param ordering
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          size: Long,
+	          ordering: Ordering): SMA =
+		apply(measureCol,size,ordering,trailing)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param outputCol
+	 * @param size
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          outputCol: String,
+	          size:Long): SMA =
+		apply(measureCol,outputCol,size,row)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param size
+	 * @return
+	 */
+	def apply(measureCol: String,
+	          size:Long): SMA =
+		apply(measureCol,size,row)
+
+	/**
+	 *
+	 * @param measureCol
+	 * @param outputCol
+	 * @param size
+	 * @param ordering
+	 * @param alignment
+	 * @param includesNow
+	 * @return
+	 */
+	def apply( measureCol: String,
+	           outputCol: String,
+	           size: Long,
+	           ordering: String,
+	           alignment: String,
+	           includesNow: Boolean): SMA =
+		apply(measureCol, outputCol, size, Orderings.withName(ordering), WindowAlignments.withName(alignment), includesNow)
```

Comment:
Can we have an option for no measure columns specific? By default all numeric columns should have this transformation applied.

Reply:
For reference, here's the pandas version : 

df.ewm(com=0.5).mean()

Reply:
Ok - that's a bigger change, since it entails applying a transform to multiple columns. We *can* do that, of course, I'm just thinking how best to implement....